### ISSUE TITLE
[Test Proxy] Compress recordings size

### ISSUE THREAD

**Issue by HarshaNalluru** [Azure/azure-sdk-tools#2724]
Certain old recording from JS repo
![image](https://user-images.githubusercontent.com/10452642/153098421-858926ea-0559-459d-b6d4-f7bf86508604.png)

Same recording with the test-proxy based recorder
![image](https://user-images.githubusercontent.com/10452642/153098483-08f0cb31-7c0d-4f58-9c36-4072088de3f7.png)

Size of `recording_pdf_file_stream` went from 312KB to 524KB.

And re-recording the tests in https://github.com/Azure/azure-sdk-for-js/pull/20264 have added 100K more lines
![image](https://user-images.githubusercontent.com/10452642/153098775-6c662f20-dae4-467a-a255-05775b466536.png)

It might be beneficial to compress the recordings before saving them - "uglifying the JSON" of sorts to save a ton of space.

/cc @witemple-msft

**Comment by witemple-msft** [comment]
Yeah, we might end up with hundreds of megabytes or gigabytes of space characters in the repo if we migrate everything to a version of the JSON that is formatted for readability.

We can always format the files ad-hoc in the editor if needed, so there's not much need to store the prettified versions at rest.
